linux_os/guide/system/software/system-tools/package_gcc_installed/rule.yml

```diff
@@ -0,0 +1,16 @@
+documentation_complete: true
+
+title: Ensure GCC compiler is installed
+
+description: The gcc package should be installed.
+
+rationale: GCC can compile programs to binary executables
+
+template: package_installed
+template_data:
+    PKGNAME: gcc
+    EVR: 0:9.2.1-1
+template_backends:
+    bash: off
```

Comment:
All content around the template should be sub-keys of template. And really, I don't think that we need to have the `template_data` or `template_backends` keys at all. A better and cleaner implementation of the template key would be:
```
template: 
    name: package_installed
    pkgname: gcc
    evr: 0:9.2.1-1
    bash: off
```
or
```
template: 
    name: package_installed
    vars:
        pkgname: gcc
        evr: 0:9.2.1-1
    bash: off
````

All keys should be lowercase.

Reply:
I like the idea to have just one key for all of the template-related stuff. However, sub-keys for name, vars and backends are good, mainly because a person who is not familiar with the content will immediately understand what's going on, that `bash` is not a variable, but it switches `bash` remediation off, and that `name` is special.
It has also pleasant side-effects that data are better organized, parsing code will not have to have to deduce what category a key belongs to, and there won't be key name conflicts.

Reply:
The current proposal is:
```
template:
    name: template_name
    vars:
        param_name@rhel7: value1
        param_name@rhel8: value2
        other_param: other_value
    backends:
        ansible:  off
        bash: on # default (implicit)
```

eg.

```
template: 
    name: services_enabled
    vars:
        package@rhel: audit
        package@ubuntu: auditd
        daemon_name: auditd
    backends:
        ansible: off
````

Reply:
Regarding the request that the keys should be lowercase. In this PR they're uppercase because I pass they directly to Jinja which processes the templates and in templates we now use uppercase Jinja variables everywhere. We have briefly discussed that with @matejak an d @yuumasato and I the solution will be to use lowercase in Jinja and batch convert all the templates to use lowercase variables. That change can be a part of commit that fills the data to rule YAMLs.

Reply:
Not seeing a need for the `backends` key at all since `ansible: off` relates to the template key itself making `backends` superfluous.... unless we need to add `ansible@rhel7` to the key.

Reply:
The separate `backends` key makes the code in `ssg/templates.py` that uses it slightly easier and less error-prone, which was @matejak 's point I believe. On the other hand it forces the person that uses it in `rule.yml` to write one more key, but it will happen rarely, because by default we plan to generate every type of content, so it will be used only if they want to not generate something (eg. turn off Ansible remediations for a specific rule). I'm fine with either way. We just need to decide. @yuumasato @matejak

Reply:
Still don't see how it makes it error prone since building all templates will be the default unless something like `ansible: off` is added.

Reply:
We will be fine with or without `backends` key.  I don't have a strong opinion for either side.

Reply:
Let's have the `backends` key there, as it will make the template definition more readable, and it will allow the parsing code to be a bit simpler. We will need it only for a very small amount of rules anyway.